### PR TITLE
feat(cli): Custom debounce limit

### PR DESCRIPTION
**Pull request by byawitz**
## What does this PR do?

(Provide a description of what this PR does.)

## Test Plan

(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work.)

## Related PRs and Issues

* https://github.com/appwrite/sdk-for-cli/issues/135

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

(Write your answer here.)

### CODE REVIEW

File: templates/cli/lib/commands/push.js.twig [hunk 1]
```diff
@@ -90,11 +91,13 @@ const awaitPools = {
...
-        if (steps > 1 && iteration === 1) {
-            pollMaxDebounces *= steps;
+        if (pollMaxDebounces === POLL_DEFAULT_VALUE) {
+            let steps = Math.max(1, Math.ceil(total / STEP_SIZE));
+            if (steps > 1 && iteration === 1) {
+                pollMaxDebounces *= steps;
 
-            log('Found a large number of attributes, increasing timeout to ' + (pollMaxDebounces * POLL_DEBOUNCE / 1000 / 60) + ' minutes')
+                log('Found a large number of attributes, increasing timeout to ' + (pollMaxDebounces * POLL_DEBOUNCE / 1000 / 60) + ' minutes')
+            }
```
Comment (stnguyen90) [inline templates/cli/lib/commands/push.js.twig:100]: So we're only going to log if the developer doesn't specify a custom debounce? Why is that?

File: templates/cli/lib/commands/push.js.twig [hunk 2]
```diff
@@ -1410,6 +1427,7 @@ push
     .command("collection")
     .alias("collections")
     .description("Push collections in the current project.")
+    .option(`-d, --debounce <maxDebounces>`, `Max number of debounces, default: 30.`)
```
Comment (stnguyen90) [inline templates/cli/lib/commands/push.js.twig:1430]: This description is pretty confusing for people because they don't know how the push code works. How can we rephrase this to be clearer?

File: templates/cli/lib/commands/push.js.twig (hunk 1 above)
Comment (byawitz) [reply templates/cli/lib/commands/push.js.twig:100]: If the developer specified the debounce limit, it will be used, and then it doesn't make sense to notify about it as it will be used anyway.

Comment (byawitz) [reply templates/cli/lib/commands/push.js.twig:100]: If the developer specified the debounces limit, then there's no need to check if we're increasing the timeout as we won't increase it but set it to the manually one

File: templates/cli/lib/commands/push.js.twig (hunk 2 above)
Comment (byawitz) [reply templates/cli/lib/commands/push.js.twig]: Is this better?

**Review by stnguyen90 — APPROVED** [review]

File: templates/cli/lib/commands/push.js.twig [hunk 3]
```diff
@@ -1410,6 +1427,7 @@ push
     .command("collection")
     .alias("collections")
     .description("Push collections in the current project.")
+    .option(`-a, --attempts <numberOfAttempts>`, `Max number of attempts before timing out collection attribute operations, default: 30.`)
```
Comment (stnguyen90) [inline templates/cli/lib/commands/push.js.twig:1430]: attempts is definitely much better than debounce. The only thing is it's not just for attribute operations; it's for indexes too. So, maybe just:

Max number of attempts before timing out

**Review by christyjacob4 — APPROVED** [review]
